### PR TITLE
add scroll and zoom option to map

### PR DESCRIPTION
**Pull request by moontrip**
@bobular and @dmfalke

This is an implementation to add a checkbox of scroll and zoom as suggested by Dave which is related to this issue,  https://github.com/VEuPathDB/web-components/issues/330

As a test and demo purpose, this is made as a draft PR. A story can be found at Map -> General -> Scroll and Zoom for testing. 

scrollAndZoom1

### CODE REVIEW

File: src/map/MapVEuMap.tsx [hunk 1]
```diff
@@ -163,6 +163,8 @@ export interface MapVEuMapProps {
   showNoDataOverlay?: boolean;
   /** Whether to show the Scale in the map */
   showScale?: boolean;
+  /** is map scroll and zoom allowed? */
+  isMapScroll?: boolean;
```
Comment (dmfalke) [inline src/map/MapVEuMap.tsx:167]: I'm not sure this is the clearest name for this property. What about something like `scrollingEnabled`?

**Review by dmfalke — APPROVED** [review]
This looks good to me!

File: src/map/MapVEuMap.tsx (hunk 1 above)
Comment (moontrip) [reply src/map/MapVEuMap.tsx:167]: thank you for your review and suggestion, @dmfalke. Yep it would work 👍 Will change

Comment (moontrip) [reply src/map/MapVEuMap.tsx]: @dmfalke changed as suggested 👍

**Comment by moontrip** [comment]
@bobular @dmfalke I wonder if we need this or just leave the map as is

**Comment by bobular** [comment]
Sorry I didn't see this as I wasn't a reviewer!
I will have a play tomorrow.

**Comment by moontrip** [comment]
sorry @bobular At that time I made this draft PR, I just wanted to make sure whether my implementation was what Dave wanted. I just added you as a reviewer too

File: src/map/MapVEuMap.tsx [hunk 2]
```diff
@@ -186,6 +188,7 @@ function MapVEuMap(props: MapVEuMapProps, ref: Ref<PlotRef>) {
     showSpinner,
     showNoDataOverlay,
     showScale = true,
+    scrollingEnabled,
```
Comment (bobular) [inline src/map/MapVEuMap.tsx:191]: Hi @moontrip - please could you make this default to true, then I would like to approve and merge this PR and use the new prop for the FSM minimap.
I might do this before you log on for the day. Let's keep in touch via Slack.
```suggestion
    scrollingEnabled = true,
```

Comment (bobular) [reply src/map/MapVEuMap.tsx:191]: Actually, I think I was trying to kill two birds with one stone.

Disabling scroll-wheel functionality isn't enough for the minimap.

I'll add an `interactive` (defaults to true) prop for MapVEuMap in my fsm-minimap branch.

Still happy for this scrollingEnabled prop to be added though :-)

Comment (moontrip) [reply src/map/MapVEuMap.tsx]: @bobular this took some time to resolve merge conflicts, but I have applied your suggestion to the lasted commit 👍

**Review by bobular — APPROVED** [review]
Cool, thanks!

**Pull request merged**